File: SelfPrec/HelixMatrix.cpp
/*
  螺旋矩阵的构造，需要考虑维度的奇偶性、方向、发散或者收敛、开始位置
  方向和发散性在某种角度看是对称的
  http://www.cnblogs.com/eshizhan/archive/2010/06/01/1749013.html
*/
#include <iostream>
using namespace std;
void simplest(int n);
int main(){
  cout << "Input the Number:\n";
  int number(0);
  cin >> number;
  cout << "do generate the helix matrix:\n";
  simplest(number);
  return 0;
}
void simplest(int n){
  int matrix[100][100] ={0};
  int i,j;
  for(i=0;i<n;i++)
    for(j = 0;j<n;j++) 
      matrix[i][j] =0;
  int rolls(0),num = 1;
  while(num <= n*n){
    for(i=rolls;i<n-rolls;i++)
       matrix[rolls][i] = num ++;
    for(i=rolls+1;i<n-rolls;i++)
       matrix[i][n-rolls-1] = num ++;
    for(i=n-rolls-2;i>=rolls;i--)
       matrix[n-rolls-1][i]= num ++;
    for(i=n-rolls-2;i>rolls;i--)
       matrix[i][rolls] = num ++;
    rolls ++;
  }
  for(i=0;i<n;i++){
    for(j=0;j<n;j++){
      cout<<matrix[i][j]<<"  ";
    }
    cout<<endl;
  }
}

File: SelfPrec/HelixMatrix.java
/*
  螺旋矩阵的构造，需要考虑维度的奇偶性、方向、发散或者收敛、开始位置
  方向和发散性在某种角度看是对称的
  http://www.cnblogs.com/eshizhan/archive/2010/06/01/1749013.html
*/
import java.util.Scanner;
class Arrays{
    int size;
    int array[][];
    public Arrays(int n){
        size = n;
        array = new int[size][size];
        for(int i =0;i<size;i++)
            for(int j=0;j<size;j++)
                 array[i][j] = 0;
    }
    public void Generate(){
        int nr=0,nd=0,nl=0,nu=0,i=1,j=0;
        while(i<size*size){
            for(j=0;j<(size-2*nr);j++){
                array[nr][j+nu]=i++;
            }
	    nr++;
            for(j=0;j<(size-2*nd-1);j++){
                array[j+nr][size-nd-1]=i++;
            }
            nd++;
            for(j=0;j<(size-2*nl-1);j++){
                array[size-1-nl][size-1-nd-j]=i++;
            }
            nl++;
            for(j=0;j<(size-2*(nu+1));j++){
                array[size-1-nl-j][nu]=i++;
            }
            nu++;
        }
    }
}
